src/d3-ext/profile.js

```diff
@@ -0,0 +1,320 @@
+/**
+ * @fileoverview Provides a D3js component to be used to draw a elevation
+ * profile chart.
+ *
+ * Example usage:
+ *
+ * var selection = d3.select('#element_id');
+ * var profile = ngeo.profile().hover(function(point) {
+ *   console.log(point.x, point.y);
+ * }).out(function() {console.log("out");});
+ * selection.datum(data).call(profile);
+ *
+ *
+ * The selection data to be used should look like the following:
+ *
+ * [
+ *     {
+ *         "y": 199340,
+ *         "values": {"mnt": 788.7},
+ *         "dist": 0.0,
+ *         "x": 541620
+ *     }, ...
+ * ]
+ *
+ */
+goog.provide('ngeo.profile');
+
+
+/**
+ *
+ * @return {Object}
+ * @export
+ *
+ */
+ngeo.profile = function() {
+
+  /**
+   * The values for margins around the chart defined in pixels.
+   */
+  var margin = {top: 10, right: 10, bottom: 30, left: 40};
+
+  /**
+   * Method to get the coordinate in pixels from a distance.
+   */
+  var bisectDistance = d3.bisector(function(d) { return d['dist']; }).left;
+
+  /**
+   * Distance units. Either 'm' or 'km'.
+   */
+  var units;
+
+  /**
+   * Factor to determine whether to use 'm' or 'km'.
+   */
+  var xFactor;
+
+  /**
+   * @type {function(Object)}
+   * Hover callback function.
+   */
+  var hover = function() {};
+
+  /**
+   * @type {function()}
+   * Out callback function.
+   */
+  var out;
+
+  /**
+   * The color to be used for filling the area.
+   * Can be overriden using the '.area' CSS selector.
+   * @type {string}
+   */
+  var FILL_COLOR = 'rgba(222, 222, 222, 0.5)';
+
+  /**
+   * The color to be used the line stroke.
+   * Can be overriden using the '.line' CSS selector.
+   * @type {string}
+   */
+  var STROKE_COLOR = '#F00';
+
+  /**
+   * @type {Object}
+   */
+  var svg;
+
+  /**
+   * D3 x scale.
+   */
+  var x;
+
+  /**
+   * D3 y scale.
+   */
+  var y;
+
+  var profile = function(selection) {
+    selection.each(function(data) {
+
+      var width = this.getBoundingClientRect().width -
+          margin.right - margin.left;
+      x = d3.scale.linear().range([0, width]);
+
+      var height = this.getBoundingClientRect().height -
+          margin.top - margin.bottom;
+      y = d3.scale.linear().range([height, 0]);
+
+      var xAxis = d3.svg.axis().scale(x).orient('bottom');
+      var yAxis = d3.svg.axis()
+          .scale(y)
+          .orient('left');
+
+      var area = d3.svg.area()
+          .x(function(d) { return x(d['dist']); })
```

Comment:
A custom function could be used here to extract the dist value from the data row.
`extractorFn.getDistance(data[i]);`
The other values: `x, y and z` could be extracted in the same way.